moabb/evaluations/splitters.py

```diff
@@ -0,0 +1,133 @@
+    >>> csess = WithinSessionSplitter(n_folds=2)
+    >>> csess.get_n_splits(metadata)
+    4
+    >>> for i, (train_index, test_index) in enumerate(csess.split(y, metadata)):
+    ...    print(f"Fold {i}:")
+    ...    print(f"  Train: index={train_index}, group={subjects[train_index]}, session={sessions[train_index]}")
+    ...    print(f"  Test:  index={test_index}, group={subjects[test_index]}, sessions={sessions[test_index]}")
+    Fold 0:
+      Train: index=[4 7], group=[1 1], session=['E' 'E']
+      Test:  index=[5 6], group=[1 1], sessions=['E' 'E']
+    Fold 1:
+      Train: index=[5 6], group=[1 1], session=['E' 'E']
+      Test:  index=[4 7], group=[1 1], sessions=['E' 'E']
+    Fold 2:
+      Train: index=[2 3], group=[1 1], session=['T' 'T']
+      Test:  index=[0 1], group=[1 1], sessions=['T' 'T']
+    Fold 3:
+      Train: index=[0 1], group=[1 1], session=['T' 'T']
+      Test:  index=[2 3], group=[1 1], sessions=['T' 'T']
+    """
+
+    def __init__(
+        self,
+        cv=StratifiedKFold,
+        custom_cv=False,
+        n_folds: int = 5,
+        random_state: int = 42,
+        shuffle: bool = True,
+        calib_size: int = None,
+    ):
+        self.n_folds = n_folds
+        self.shuffle = shuffle
+        self.random_state = check_random_state(random_state) if shuffle else None
+        self.cv = cv
+        self.calib_size = calib_size
+        self.custom_cv = custom_cv
+
+    def get_n_splits(self, metadata):
+        num_sessions_subjects = metadata.groupby(["subject", "session"]).ngroups
+        return (
+            self.cv.get_n_splits(metadata)
+            if self.custom_cv
+            else self.n_folds * num_sessions_subjects
+        )
+
+    def split(self, y, metadata, **kwargs):
+        all_index = metadata.index.values
+        subjects = metadata["subject"].unique()
+
+        # Shuffle subjects if required
+        if self.shuffle:
+            self.random_state.shuffle(subjects)
+
+        for i, subject in enumerate(subjects):
+            subject_mask = metadata.subject == subject
+            subject_indices = all_index[subject_mask]
+            subject_metadata = metadata[subject_mask]
+            sessions = subject_metadata.session.unique()
+
+            # Shuffle sessions if required
+            if self.shuffle:
+                self.random_state.shuffle(sessions)
+
+            for j, session in enumerate(sessions):
+                session_mask = subject_metadata.session == session
+                indices = subject_indices[session_mask]
+                group_y = y[indices]
+
+                # Handle custom splitter
+                if isinstance(self.cv(), PseudoOnlineSplit):
+                    splitter = self.cv(calib_size=self.calib_size)
+                    for calib_ix, test_ix in splitter.split(
+                        indices, group_y, subject_metadata[session_mask]
+                    ):
+                        yield calib_ix, test_ix
```

Comment:
As `PseudoOnlineSplit` inherits from `BaseCrossValidator`, it should have the same behavior as any other cross validators. In other words, it should be implemented such that this line is changed into:
```suggestion
                        yield indices[calib_ix], indices[test_ix]
```